### DISCUSSION TITLE
Deploying on local nuc

### DISCUSSION THREAD

**Discussion by crazzywizard** [ponder-sh/ponder#555] [Question]
I would like to deploy ponder application locally using my nuc. I was able to get the ports open and route domain traffic to http://localhost:42069. However I am running into an issue where it fails to retrieve data from devices outside of local nuc hosting the service. I am using nginx to handle traffic routing. When I go to the domain on other devices and inspect the page I see the post request url is to localhost and not to the domain, which makes me think I am missing some config. Any ideas or help would be greatly appreciated.

**Comment by typedarray** [comment]
Hi! Thanks for opening. Were you using the GraphiQL explorer when you noticed this behavior? I just opened a PR (https://github.com/ponder-sh/ponder/pull/559) that adds a new `--hostname` CLI option which you can use to override the default of `0.0.0.0`. That PR also should fix an issue where the GraphiQL page was making requests to the wrong URL.

Preview of docs for new CLI options:
https://ponder-docs-49i7kk8j9-cantrip.vercel.app/docs/api-reference/ponder-cli#dev

<img width="942" alt="Screen Shot 2024-01-05 at 5 28 18 PM" src="https://github.com/ponder-sh/ponder/assets/90073088/1a406344-0adb-41f8-95f2-d5987b8f4511">
